File: app/src/main/java/com/example/krystianwsul/organizator/utils/time/NormalTime.java
package com.example.krystianwsul.organizator.utils.time;

import android.support.v4.util.Pair;

import com.example.krystianwsul.organizator.domainmodel.CustomTime;

public class NormalTime implements Time {
    private final HourMinute mHourMinute;

    public NormalTime(HourMinute hourMinute) {
        mHourMinute = hourMinute;
    }

    public NormalTime(int hour, int minute) {
        mHourMinute = new HourMinute(hour, minute);
    }

    public String getName() {
        return mHourMinute.getHour() + ":" + mHourMinute.getMinute();
    }

    public HourMinute getHourMinute(DayOfWeek dayOfWeek) {
        return mHourMinute;
    }

    public HourMinute getHourMinute() {
        return mHourMinute;
    }

    public String toString() {
        return mHourMinute.toString();
    }

    @Override
    public Pair<CustomTime, HourMinute> getPair() {
        return new Pair<>(null, mHourMinute);
    }

    @Override
    public TimePair getTimePair() {
        return new TimePair(null, mHourMinute);
    }
}

File: common/src/commonMain/kotlin/time/NormalTime.kt
package com.krystianwsul.common.time

data class NormalTime(val hourMinute: HourMinute) : Time {

    override val timePair get() = TimePair(null, hourMinute)

    constructor(hour: Int, minute: Int) : this(HourMinute(hour, minute))

    override fun getHourMinute(dayOfWeek: DayOfWeek) = hourMinute

    override fun toString() = hourMinute.toString()
}
